lib/elixir/lib/integer.ex

```diff
@@ -64,6 +64,49 @@ defmodule Integer do
   """
   defguard is_even(integer) when is_integer(integer) and (integer &&& 1) == 0
 
+  @doc """
+  Computes `base` raised to power of `exponent`.
+
+  Both `base` and `exponent` must be integers.
+  The exponent must be zero or positive.
+
+  See `Float.pow/2` for exponentiation of negative
+  exponents as well as floats.
+
+  ## Examples
+
+      iex> Integer.pow(2, 0)
+      1
+      iex> Integer.pow(2, 1)
+      2
+      iex> Integer.pow(2, 10)
+      1024
+      iex> Integer.pow(2, 11)
+      2048
+      iex> Integer.pow(2, 64)
+      0x10000000000000000
+
+      iex> Integer.pow(3, 4)
+      81
+      iex> Integer.pow(4, 3)
+      64
+
+      iex> Integer.pow(-2, 3)
+      -8
+      iex> Integer.pow(-2, 4)
+      16
+
+  """
+  def pow(base, exponent) when is_integer(base) and is_integer(exponent) and exponent >= 0 do
```

Comment:
```suggestion
  """
  @doc since: "1.12.0"
  @spec pow(integer, pos_integer) :: integer
  def pow(base, exponent) when is_integer(base) and is_integer(exponent) and exponent >= 0 do
```